### DISCUSSION TITLE
dotnet wasm runtime in Edge

### DISCUSSION THREAD

**Discussion by dmitrifaleev** [dotnet/runtime#109686] [General]
I couldn't find if this came up before. Why isn't the dotnet wasm runtime included in Edge? with CDN caching going away, and lots of effort to improve loading times (e.g. Blazor) it seems like including the runtime would be beneficial to a lot of users.

**Comment by maraf** [comment]
It would probably be challenging in some ways, what comes to my mind now
- What versions should it include? All currently supported with fallback to download if the required one by the app is not included?
- AOT creates a specific version of dotnet wasm containing AOTed code based on the use in the application
- Trimming creates a specific version of all managed dlls 
- Linking any native code, like skiasharp or anything else, creates a specific version of dotnet wasm
- The same is true for opting in/out of wasm features like SIMD or exception handling
- Download time is payed only once per application deployment (until it gets cached in the browser). In all the other cases only the time for reading the file from disk cache and doing the wasm instantiation is payed

**Reply by dmitrifaleev** [reply]
I can see where this could get complex. The option to include  supported versions and fallback to download would be nice to improve users first impression imo
